extension/externalsubscriber/grpchandler.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package externalsubscriber
+
+import (
+	"net"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	pb "github.com/ava-labs/hypersdk/proto/pb/externalsubscriber"
+)
+
+type GRPCHandler struct {
+	gRPCServer *grpc.Server
+	listener   net.Listener
+	tcpPort    string
+	log        logging.Logger
+}
+
+func NewGRPCHandler(e *ExternalSubscriberServer, log logging.Logger, tcpPort string) (*GRPCHandler, error) {
+	lis, err := net.Listen("tcp", tcpPort) // #nosec G102
+	if err != nil {
+		return nil, err
+	}
+	log.Info("listening to tcp port", zap.String("address", tcpPort))
+
+	s := grpc.NewServer()
+	pb.RegisterExternalSubscriberServer(s, e)
+	return &GRPCHandler{
+		gRPCServer: s,
+		listener:   lis,
+		tcpPort:    tcpPort,
+		log:        log,
+	}, nil
+}
+
+func (g *GRPCHandler) Start() {
+	go func() {
+		if err := g.gRPCServer.Serve(g.listener); err != nil {
```

Comment:
Looks like we ignore this by default in AvalancheGo https://github.com/ava-labs/avalanchego/blob/82bf033c3fb495d9441b752e4f168b3dc58513c8/vms/rpcchainvm/grpcutils/server.go#L101